cpp/dolfinx/fem/utils.h

```diff
@@ -682,6 +686,117 @@ pack_coefficients(const Form<T>& form, IntegralType integral_type, int id)
   return {std::move(c), cstride};
 }
 
+/// Create Expression from UFC
+template <typename T>
+fem::Expression<T> create_expression(
+    const ufcx_expression& expression,
+    const std::vector<std::shared_ptr<const fem::Function<T>>>& coefficients,
+    const std::vector<std::shared_ptr<const fem::Constant<T>>>& constants,
+    const std::shared_ptr<const fem::FunctionSpace> argument_function_space = nullptr,
+    std::shared_ptr<const mesh::Mesh> mesh = nullptr)
+{
+  if (expression.rank > 0 and !argument_function_space)
+  {
+    throw std::runtime_error(
+        "Expression has Argument but no Argument function space was provided");
+  }
+
+  if (argument_function_space)
+  {
+    if (!mesh)
+      mesh = argument_function_space->mesh();
+    if (argument_function_space->mesh()->id() != mesh->id())
```

Comment:
No need to use `id` to check. Can use `if (argument_function_space->mesh() != mesh)`.